### PR TITLE
Filter buttons: Update the DOM with JS to work around Webflow restrictions 

### PR DESCRIPTION
**Pull request by cameralibre**
The figma design for https://the-lumery-staging.webflow.io/tomorrow-projects and https://the-lumery-staging.webflow.io/projects requires the 'All projects' button to be inline with the other tag buttons (these are all visually a variant of secondary buttons, but semantically radio buttons)

I'm using [Finsweet's CMS Filter](https://finsweet.com/attributes/cms-filter) for the functionality of this filter, which uses a CMS collection to manage the tags. You're not allowed to put anything else into a collection list though.

image

So this hack will shove the 'All projects' button into the right place in the DOM.

It takes a while for the fonts to load, so there's a bit of layout shift on the header on first load, it's not too bad in the hero heading as it has the chaotic animation which hides it a bit, but the filter buttons jump around considerably. I decided to wait until after all this button-shifty-jiggery-pokery was finished, and the fonts have likely loaded, and the hero animation is finishing, before fading them in.

### CODE REVIEW

**Comment by cameralibre** [comment]
whoops, I didn't merge my PRs in order... you only need to review the [_first_ commit](https://github.com/springload/webflow-dev-workflow/pull/13/commits/deda2f04d5e35503d6fdf4fc12b1324aefab9e74) in this PR rather than the whole thing

File: client-projects/the-lumery/filterButtons.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  );
+  if (!resetButton) {
+    throw new Error(
+      `Reset button element not found: ${FILTER_RESET_BUTTON_SELECTOR}`,
+    );
+  }
+  const filterButtonList: HTMLElement | null = document.querySelector(
+    FILTER_BUTTON_LIST_SELECTOR,
+  );
+  if (!filterButtonList) {
```
Comment (cameralibre) [inline client-projects/the-lumery/filterButtons.ts:18]: I've just realised that I can do the [`instanceOf HTMLElement` check](https://github.com/springload/webflow-dev-workflow/pull/14#discussion_r1382635964) here as well to tidy things up a bit.

File: client-projects/the-lumery/filterButtons.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  filterButtonList.insertBefore(
+    resetButton,
+    filterButtonList.firstElementChild,
+  );
+
+  resetButton.style.opacity = "1";
+
+  filterButtonList.animate([{ opacity: 0 }, { opacity: 1 }], {
+    duration: 400,
+    delay: 900, // allow fonts to load and the header animation to finish
```
Comment (liamjohnston) [inline client-projects/the-lumery/filterButtons.ts:33]: How much of this 900 is waiting for fonts (vs waiting for the header animation)? Don't love the magic number-ness of it, but I guess that ensuring that the font gets loaded is both tricky and unnecessarily blocking so 🤷

**Review by liamjohnston — APPROVED** [review]

File: client-projects/the-lumery/filterButtons.ts (hunk 2 above)
Comment (cameralibre) [reply client-projects/the-lumery/filterButtons.ts:33]: Fonts get loaded at around 500/600ms on my machine and connection. I _could_ do some check for the fonts before fading in the buttons, but I don't want to have new stuff appearing on screen while the header is animating, so I extended the delay to fade in at the end of the animation

**Pull request merged**